docs/integrations/data-ingestion/clickpipes/postgres/joins_and_denormalization.md

```diff
@@ -0,0 +1,278 @@
+
+- **Use subqueries or CTE for filtering**: Modify JOINs as subqueries where you filter tables within the subquery before passing them to the planner. This is usually unnecessary, but it's sometimes worth trying. Below is an example of a JOIN query using a sub-query.
+
+```sql
+-- Use a subquery to reduce the number of rows to join
+SELECT
+    t.id AS UserId,
+    t.displayname,
+    t.views,
+    COUNTDistinct(multiIf(c.id != 0, c.id, NULL)) AS CommentsCount
+FROM (
+    SELECT id, displayname, views
+    FROM users
+    ORDER BY views DESC
+    LIMIT 10
+) t
+LEFT JOIN comments c ON t.id = c.userid
+GROUP BY t.id, t.displayname, t.views
+ORDER BY t.views DESC
+SETTINGS final=1;
+```
+
+-  **Optimize Ordering Keys**: Consider including JOIN columns in the `Ordering Key` of the table. For more details, refer to the page on [Ordering Key](/integrations/clickpipes/postgres/ordering_keys).
+
+- **Use Dictionaries for dimension tables**: Consider creating a [dictionary](/sql-reference/dictionaries) from a table in ClickHouse to improve lookup performance during query execution. This [documentation](/dictionary#speeding-up-joins-using-a-dictionary) provides an example of how to use dictionaries to optimize JOIN queries with a StackOverflow dataset.
+
+- **JOIN algorithms**: ClickHouse offers various [algorithms](/guides/joining-tables) for joining tables, and selecting the right one depends on the specific use case. Below are two examples of JOIN queries using different algorithms tailored to distinct scenarios: in the first case, the goal is to reduce memory usage, so the partial_merge algorithm is used, while in the second case, the focus is on performance, and the parallel_hash algorithm is used. Note the difference in memory used.
+
+```sql
+-- Use partial merge algorithm
+SELECT
+    sum(p.viewcount) AS viewcount,
+    p.owneruserid AS user_id,
+    u.displayname AS display_name
+FROM posts AS p
+FINAL
+LEFT JOIN users AS u
+FINAL ON (u.id = p.owneruserid) AND (u._peerdb_is_deleted = 0)
+WHERE (p.owneruserid > 0) AND (p._peerdb_is_deleted = 0)
+GROUP BY
+    user_id,
+    display_name
+ORDER BY viewcount DESC
+LIMIT 10
+FORMAT `NULL`
+SETTINGS join_algorithm = 'partial_merge'
+
+10 rows in set. Elapsed: 7.202 sec. Processed 60.42 million rows, 1.83 GB (8.39 million rows/s., 254.19 MB/s.)
+Peak memory usage: 1.99 GiB.
+
+-- Use parallel hash algorithm
+SELECT
+    sum(p.viewcount) AS viewcount,
+    p.owneruserid AS user_id,
+    u.displayname AS display_name
+FROM posts AS p
+FINAL
+LEFT JOIN users AS u
+FINAL ON (u.id = p.owneruserid) AND (u._peerdb_is_deleted = 0)
+WHERE (p.owneruserid > 0) AND (p._peerdb_is_deleted = 0)
+GROUP BY
+    user_id,
+    display_name
+ORDER BY viewcount DESC
+LIMIT 10
+FORMAT `NULL`
+SETTINGS join_algorithm = 'parallel_hash'
+
+10 rows in set. Elapsed: 2.160 sec. Processed 60.42 million rows, 1.83 GB (27.97 million rows/s., 847.53 MB/s.)
+Peak memory usage: 5.44 GiB.
+```
+
+### Denormalization {#denormalization}
```

Comment:
Again we have a dedicated guide @lio-p lets enhance the guide and link vs repeat